src/legacy/ui/public/courier/fetch/call_client.js

```diff
@@ -20,35 +20,27 @@
 import { ErrorAllowExplicitIndexProvider } from '../../error_allow_explicit_index';
 import { assignSearchRequestsToSearchStrategies } from '../search_strategy';
 import { IsRequestProvider } from './is_request';
-import { MergeDuplicatesRequestProvider } from './merge_duplicate_requests';
 import { RequestStatus } from './req_status';
 import { SerializeFetchParamsProvider } from './request/serialize_fetch_params';
 import { i18n } from '@kbn/i18n';
 
 export function CallClientProvider(Private, Promise, es, config) {
   const errorAllowExplicitIndex = Private(ErrorAllowExplicitIndexProvider);
   const isRequest = Private(IsRequestProvider);
-  const mergeDuplicateRequests = Private(MergeDuplicatesRequestProvider);
   const serializeFetchParams = Private(SerializeFetchParamsProvider);
 
   const ABORTED = RequestStatus.ABORTED;
-  const DUPLICATE = RequestStatus.DUPLICATE;
 
   function callClient(searchRequests) {
     const maxConcurrentShardRequests = config.get('courier:maxConcurrentShardRequests');
     const includeFrozen = config.get('search:includeFrozen');
 
-    // merging docs can change status to DUPLICATE, capture new statuses
-    const searchRequestsAndStatuses = mergeDuplicateRequests(searchRequests);
-
-    // get the actual list of requests that we will be fetching
-    const requestsToFetch = searchRequestsAndStatuses.filter(isRequest);
```

Comment:
What's the reason for removing this variable and this filtering logic? I think this will allow us to exit early if the `searchRequests` is populated with non-SearchRequest values, e.g. an array of `ABORTED` statuses.

Reply:
Good point, I'll revert this change